app/src/main/java/com/github/lookupgroup27/lookup/model/map/Renderer.kt

```diff
@@ -24,34 +24,45 @@ class Renderer : GLSurfaceView.Renderer {
 
     GLES20.glEnable(GLES20.GL_DEPTH_TEST)
 
-    // Create the shapes (Make sure you always create the shapes after the OpenGL context is
-    // created)
-    shapes =
-        listOf(
-            Star(0.0f, 0f, -1f, floatArrayOf(1.0f, 1.0f, 1.0f)),
-            Star(0.24f, 0f, -0.97f, floatArrayOf(1.0f, 1.0f, 1.0f)),
-            Star(1f, 0f, 0f, color = floatArrayOf(1.0f, 0.0f, 0.0f)),
-            Star(0f, 1f, 0f, color = floatArrayOf(0.0f, 1.0f, 0.0f)),
-            Star(0f, 0f, 1f, color = floatArrayOf(0.0f, 0.0f, 1.0f)))
+    // Initialize the SkyBox
+    skyBox = SkyBox()
+    skyBox.initialize()
   }
 
   override fun onDrawFrame(unused: GL10) {
-    // Redraw background color
-    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT)
-    GLES20.glClear(GLES20.GL_DEPTH_BUFFER_BIT)
 
-    // Draw the shapes
-    for (shape in shapes) shape.draw(camera)
+    // Clear the screen
+    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT or GLES20.GL_DEPTH_BUFFER_BIT)
+
+    GLES20.glDepthMask(false) // Disable depth writing
+
+    // Use this MVP matrix to render the skybox
+    skyBox.draw(camera)
+
+    GLES20.glDepthMask(true) // Re-enable depth writing for other objects
   }
 
   override fun onSurfaceChanged(unused: GL10, width: Int, height: Int) {
     // Specify the size of the rendering window
     GLES20.glViewport(0, 0, width, height)
 
     val ratio: Float = width.toFloat() / height.toFloat()
-    // Defines a projection matrix in terms of a field of view angle, an aspect ratio, and z clip
-    // planes
-    // It helps projects in correct aspect ratio the objects in the scene
+
     camera.updateProjectionMatrix(ratio)
+
+    // Update the view matrix
+    Matrix.setLookAtM(
+        camera.viewMatrix,
+        0,
+        0f,
+        0f,
+        3f, // Eye position
+        0f,
+        0f,
+        0f, // Center position
+        0f,
+        1f,
+        0f // Up vector
+        )
```

Comment:
This code can likely be safely removed, as I tested it and it doesn’t seem to affect anything. This is because the camera view matrix is directly updated by the motion-based logic, so this part doesn’t appear to have any impact. 

Please let me know if I’m misunderstanding something, but I believe we can safely delete this section.